main.tf

```diff
@@ -1,13 +1,204 @@
-resource "random_integer" "example" {
-  count = module.this.enabled ? 1 : 0
+locals {
+  enabled                 = module.this.enabled
+  create_log_bucket       = var.bucket_id == null ? true : false
+  bucket_id               = var.bucket_id != "" ? var.bucket_id : module.ssm_patch_log_s3_bucket.bucket_id
+  default_allowed_actions = ["s3:GetObject", "s3:PutObject", "s3:PutObjectAcl", "s3:GetEncryptionConfiguration"]
+}
+
+module "ssm_patch_log_s3_bucket" {
+  count                  = local.create_log_bucket ? 1 : 0
+  source                 = "cloudposse/s3-bucket/aws"
+  version                = "0.38.0"
+  acl                    = "private"
+  versioning_enabled     = false
+  allowed_bucket_actions = var.ssm_bucket_policy != null ? [] : local.default_allowed_actions
+  policy                 = var.ssm_bucket_policy
+  context                = module.this.context
+}
+
+resource "aws_ssm_maintenance_window" "scan_window" {
+  count    = module.this.enabled ? 1 : 0
+  name     = "${module.this.id}-scan-window"
+  schedule = var.scan_maintenance_window_schedule
+  duration = var.maintenance_window_duration
+  cutoff   = var.maintenance_window_cutoff
+}
 
-  min = 1
-  max = 50000
-  keepers = {
-    example = var.example
+resource "aws_ssm_maintenance_window_task" "task_scan_patches" {
+  count            = module.this.enabled ? 1 : 0
+  window_id        = aws_ssm_maintenance_window.scan_window[0].id
+  task_type        = "RUN_COMMAND"
+  task_arn         = "AWS-RunPatchBaseline"
+  priority         = var.task_scan_priority
+  service_role_arn = var.service_role_arn
+  max_concurrency  = var.max_concurrency
+  max_errors       = var.max_errors
+
+  targets {
+    key    = "WindowTargetIds"
+    values = aws_ssm_maintenance_window_target.target_scan.*.id
+  }
+
+  task_invocation_parameters {
+    run_command_parameters {
+      parameter {
+        name   = "Operation"
```

Comment:
should these be parameterized in variables so the module is more generic?

Reply:
they could be, should I do it ? ( generally those are the values you want)

Reply:
if they (almost) never change, ok to keep them there.
@jamengual you sure that other people will not want to use diff values?

Reply:
I can't be sure of what people will like to do but these defaults are most probably what most people will need